arelle/plugin/validate/ROS/rules/ros.py

```diff
@@ -354,9 +354,9 @@ def rule_ros19(
     message: str | None = None
     modelObjects = set()
     monetaryFacts = val.modelXbrl.factsByDatatype(False, qnXbrliMonetaryItemType)
-    monetaryUnits = set(fact.unit.value for fact in monetaryFacts)
+    monetaryUnits = set(fact.unit.value if fact.unit and fact.unit.value else None for fact in monetaryFacts)
```

Comment:
Why `and fact.unit.value`?

It seems like this should exclude the unit altogether rather than produce a `None` (the user will already separately get an error for a missing unit on a monetary fact):
```
set(fact.unit.value for fact in monetaryFacts if fact.unit)
```